index.html

```diff
@@ -2622,178 +2649,285 @@ <h3><dfn>Writing or pushing content</dfn></h3>
               <li>
-                If |record|'s <a>id</a> is not `undefined`:
-                <ul>
-                  <li>
-                    Let |identifier| be |record|'s <a>id</a>.
-                  </li>
-                  <li>
-                    Set |ndef|'s <a>IL field</a> to `1`.
-                  </li>
-                  <li>
-                    Set |ndef|'s <a>ID LENGTH field</a> to the length of
-                    |identifier|.
-                  </li>
-                  <li>
-                    Set |ndef|'s <a>ID field</a> to |identifier|.
-                  </li>
-                </ul>
+                Otherwise, return <a>map external data to NDEF</a> given
+                |record| and |ndef|.
+              </li>
+            </ul>
+          </li>
+          <li>
+            If |record| is a payload to another <a>NDEF record</a> and if
+            running the <a>validate local type</a> steps on |record|'s
+            <a>recordType</a> returns `true`,
+            <ul>
+              <li>
+                If |record|'s <a>data</a> is of type
+                {{NDEFMessageInit}}, then return the result of running
+                the <a>create NDEF message</a> given |record|'s
+                <a>data</a>.
               </li>
               <li>
-                Add |ndef| to |output|.
+                Otherwise, return the result of running
+                <a>map local type to NDEF</a> given |record| and |ndef|.
               </li>
-              <!-- converting each record -->
-            </ol>
-          </li> <!-- converting message -->
+            </ul>
+          </li>
+          <li>
+            Otherwise, [= exception/throw =] a {{TypeError}} and abort
+            these steps.
+          </li>
         </ol>
-      </section>
+      </div>
+    </section>
+
+    <section><h3>Validating external type</h3>
+      <p>
+        The [[NFC-RTD]] specifies that external types MUST contain
+        the domain name of the issuing organization, a a colon and a type
+        name that is at least one character long.
+      </p>
+      <p>
+        The [[NFC-RTD]] specifies to omit the URN prefix “`urn:nfc:ext:`”
+        in the binary encoding of external type records, therefore
+        applications SHOULD NOT specify that when creating
+        <a>external type records</a>.
+      </p>
+      <p>
+        When the domain name is not provided, by default Web NFC uses the
+        <a href="https://html.spec.whatwg.org/multipage/origin.html#concept-origin-effective-domain">
+        effective domain of the origin</a> as domain name in external type
+        names when <a href="#mapping-external-data-to-ndef">creating external type records</a>. Therefore, applications MAY specify only the
+        colon and the external type name.
+      </p>
+      <p class="note">
+        The [[NFC-RTD]] mandates that external type names are represented
+        with the URN prefix “`urn:nfc:ext:`”, e.g. when reading
```

Comment:
maybe clarify here that that is actually not encoded - so I don;t know how much they "mandate" it. It is more considered an URL with that prefix though that is not even encoded

Reply:
In the NFC Forum spec it's absolutely normative.

Reply:
That's why the Note was made, to justify why Web NFC does not comply.